packages/client/tests/functional/too-many-instances-of-prisma-client/tests.ts

```diff
@@ -0,0 +1,42 @@
+import testMatrix from './_matrix'
+
+// @ts-ignore this is just for type checks
+declare let PrismaClient: typeof import('@prisma/client').PrismaClient
+
+testMatrix.setupTestSuite(
+  () => {
+    const oldConsoleWarn = console.warn
+    const warnings: any[] = []
+
+    beforeAll(() => {
+      console.warn = (args) => {
+        warnings.push(args)
+      }
+    })
+
+    afterAll(() => {
+      console.warn = oldConsoleWarn
+    })
+
+    test('should console warn when spawning too many instances of PrismaClient', async () => {
+      const clients: any[] = []
+      for (let i = 0; i < 15; i++) {
+        const client = new PrismaClient()
+        await client.$connect()
+        clients.push(client)
+      }
+
+      for (const client of clients) {
+        client.$disconnect()
+      }
+
+      expect(warnings.join('')).toContain('There are already 10 instances of Prisma Client actively running')
+    })
+  },
+  {
+    optOut: {
+      from: ['cockroachdb', 'mongodb', 'mysql', 'sqlite', 'sqlserver'],
+      reason: 'TODO - Test suite running in band causing issues running the console spy',
```

Comment:
Cannot figure out why this happening, but this test only works for one provider and then fails for subsequent others. Ideas? Is it fine to test this on one provider?

Reply:
I have not verified this, but I think [jest.resetModules](https://jestjs.io/docs/jest-object#jestresetmodules) will help in the `beforeAll` or `afterAll`.